animations/transition_widgets.md

```diff
@@ -0,0 +1,200 @@
+ <a href="#C4" onClick="document.img.src=imgOff2.src;">◼︎</a>
+
+ This example shows an animation using the <code>AnimatedOpacity</code> implicit widget.
+</div> -->
+
+ <!-- <br>
+ The `AnimatedOpacity` implicit widget automatically transitions a child's opacity over a given duration whenever the given opacity changes. Notice that the listeners, tickers, and other animation elements are *implied* so you don't need to add them. The `AnimatedOpacity` class extends `ImplicitlyAnimatedWidget` and those elements are already included in the `ImplicitlyAnimatedWidget` class. -->
+
+The code for the `animationexample` example is shown below. The `animationexample` widget extends the `AnimatedWidget` class and already includes the listeners, tickers, and other elements that define the animation.  
+<!-- skip -->
+{% prettify dart %}
+```Dart
+import 'package:flutter/material.dart';
+
+add the code for the transition widget
+
+```
+{% endprettify %}
+
+### < Transition animation > example 2
+<!-- The `AnimatedOpacity` implicit widget automatically transitions a child's opacity over a given duration whenever the specified opacity changes. -->
+<table cellpadding="10">
+  <tr>
+    <td style="width:20%">
+    <a href="" onMouseOver="document.MyImage1.src='images/AnimatedOpacity_blue.gif';" onMouseOut="document.MyImage1.src='images/placeholder_image.png';">
+    <img src="images/placeholder_image.png" name="MyImage1" height="180" width="160" name="MyImage1">
+    </a></td>
+    <td>
+    The <code>animationexample</code> transition widget automatically < description > .<br>Mouseover the image to view the animation.
+    </td>
+  </tr>
+</table>
+
+<!-- <a href="" onMouseOver="document.MyImage1.src='images/AnimatedOpacity_blue.gif';" onMouseOut="document.MyImage1.src='images/AnimatedOpacity_blue.png';">
+<img src="images/AnimatedOpacity_blue.png" name="MyImage1" height="180" width="160" name="MyImage1">
+</a>
+Mouseover the image to view the animation using the <code>AnimatedOpacity</code> implicit widget. -->
+
+<!-- <div>
+<img src="images/AnimatedOpacity_blue.png" name="img" height="180" width="160">
+
+ <script>
+ imgOn2 = new Image;
+ imgOff2 = new Image;
+ imgOn2.src = "images/AnimatedOpacity_blue.gif";
+ imgOff2.src = "images/AnimatedOpacity_blue.png";
+ </script>
+
+ <a href="#C4" onClick="document.img.src=imgOn2.src;">▶︎</a>
+ <a href="#C4" onClick="document.img.src=imgOff2.src;">◼︎</a>
+
+ This example shows an animation using the <code>AnimatedOpacity</code> implicit widget.
+</div> -->
+
+ <!-- <br>
+ The `AnimatedOpacity` implicit widget automatically transitions a child's opacity over a given duration whenever the given opacity changes. Notice that the listeners, tickers, and other animation elements are *implied* so you don't need to add them. The `AnimatedOpacity` class extends `ImplicitlyAnimatedWidget` and those elements are already included in the `ImplicitlyAnimatedWidget` class. -->
```

Comment:
Are those classes even used in the background? I don't know.

Reply:
The implicit widgets that exist in the Flutter SDK must extent the ImplicityAnimatedWidget so that all the elements needed for an animation to work are included - it's an abstract class.